src/array/partial_array.rs

```diff
@@ -14,6 +14,8 @@ use crate::traits::{BitVecOps, BitVecOpsMut};
 use crate::traits::{RankUnchecked, SuccUnchecked};
 use mem_dbg::*;
 
+type DenseIndex = Rank9<BitVec<Box<[usize]>>>;
```

Comment:
I had to introduce `DenseIndex` in the other one because Clippy complained about the type complexity, so I changed it here as well.